nemo/utils/export_utils.py

```diff
@@ -460,6 +461,8 @@ def replace_for_export(model: nn.Module) -> nn.Module:
     }
 
     replace_modules(model, default_Apex_replacements)
-    replace_modules(model, default_replacements)
+    # Apply CastToFloat for torch.float32
```

Comment:
This changes behavior for all other models.
Actually, casts are not needed if no autocast is involved (float32 case here) - and also we are revisiting those casts. I will submit a PR for that shortly.

Reply:
We should get #6524 merged first, and this change then can be safely reverted.

Reply:
Thanks for the review! Will keep an eye out for it to merge and test it locally in the meantime.